cmd/dra-example-kubeletplugin/state.go

```diff
@@ -173,3 +193,67 @@ func (s *DeviceState) prepareDevices(claim *resourceapi.ResourceClaim) (Prepared
 func (s *DeviceState) unprepareDevices(claimUID string, devices PreparedDevices) error {
 	return nil
 }
+
+func (pds PreparedDevices) GetDevices() []*drapbv1.Device {
+	var devices []*drapbv1.Device
+	for _, pd := range pds {
+		devices = append(devices, &pd.Device)
+	}
+	return devices
+}
+
+type PtrHasGroupVersionKind[T any] interface {
+	*T
+	GroupVersionKind() schema.GroupVersionKind
+}
+
+// GetMergedOpaqueDeviceConfig merges configs across device classes and resource claims for a given device request.
+func GetMergedOpaqueDeviceConfig[T any, TI PtrHasGroupVersionKind[T]](request string, claim *resourceapi.ResourceClaim, classes map[string]*resourceapi.DeviceClass) (T, error) {
+	// Collect all configs in reverse priority order. Device class configs have
```

Comment:
Would calling out the priority of the comments be clearer if called out in a bulleted list. Would this information also be relevant for the function documentation?

Reply:
I've updated the comment (and made it part of the godoc). Let me know what you think.

The goal is to make this generic enough that it becomes part of the in-tree helpers that ship with k8s itself.